### PR TITLE
feat!: added adapter config to TlcAdapter

### PR DESCRIPTION
**Pull request by janniclas**
-  removed KpiAdapter interface
- added configuration to TlcAdapter
- fixed missing recursive call in technical lag calculation
- added missing test cases

### CODE REVIEW

File: adapter/src/main/kotlin/de/fraunhofer/iem/kpiCalculator/adapter/KpiAdapter.kt [hunk 1]
```diff
@@ -9,32 +9,17 @@
...
-     */
-    fun transformDataToKpi(data: T): Collection<AdapterResult> = transformDataToKpi(listOf(data))
-}
 
 enum class ErrorType { DATA_VALIDATION_ERROR }
 
 sealed class AdapterResult {
-    data class Success(val rawValueKpi: RawValueKpi) : AdapterResult()
+    sealed class Success(val rawValueKpi: RawValueKpi) : AdapterResult() {
+        class Kpi(rawValueKpi: RawValueKpi) : Success(rawValueKpi)
```
Comment (AnakinRaW) [inline adapter/src/main/kotlin/de/fraunhofer/iem/kpiCalculator/adapter/KpiAdapter.kt:20]: do you want these to be data classes?

**Review by AnakinRaW — APPROVED** [review]
Do you want to remove the SupportedTools enum too?

File: adapter/src/main/kotlin/de/fraunhofer/iem/kpiCalculator/adapter/KpiAdapter.kt (hunk 1 above)
Comment (janniclas) [reply adapter/src/main/kotlin/de/fraunhofer/iem/kpiCalculator/adapter/KpiAdapter.kt:20]: yes but it was complicated.. :D 
the following code is not allowed: 
`        data class Kpi(override val rawValueKpi: RawValueKpi) : Success(rawValueKpi)`
as we may not have override modifiers in primary constructors. We also must only have val / var in primary constructors. So we can't do the pattern we do with a class `class Kpi(rawValueKpi: RawValueKpi) : Success(rawValueKpi)` and "hand down" the rawValueKpi to its superclass.

**Comment by janniclas** [comment]
SupportedTools.kt was removed

**Pull request merged**